src/Microsoft.ML.Data/Transforms/ValueMappingTransformer.cs

```diff
@@ -0,0 +1,995 @@
+        {
+            public static Column Parse(string str)
+            {
+                var res = new Column();
+                if (res.TryParse(str))
+                    return res;
+                return null;
+            }
+
+            public bool TryUnparse(StringBuilder sb)
+            {
+                Contracts.AssertValue(sb);
+                return TryUnparseCore(sb);
+            }
+        }
+
+        public sealed class Arguments
+        {
+            [Argument(ArgumentType.Multiple | ArgumentType.Required, HelpText = "New column definition(s) (optional form: name:src)", ShortName = "col", SortOrder = 1)]
+            public Column[] Column;
+
+            [Argument(ArgumentType.AtMostOnce, HelpText = "The name of the column containing the keys", ShortName = "key")]
+            public string KeyColumn;
+
+            [Argument(ArgumentType.AtMostOnce, HelpText = "The name of the column containing the values", ShortName = "value")]
+            public string ValueColumn;
+
+            [Argument(ArgumentType.Multiple, HelpText = "The data loader", NullName = "<Auto>", SignatureType = typeof(SignatureDataLoader))]
+            public IComponentFactory<IMultiStreamSource, IDataLoader> Loader;
+
+            [Argument(ArgumentType.AtMostOnce, IsInputFileName = true, HelpText = "The data file containing the terms", ShortName = "data", SortOrder = 2)]
+            public string DataFile;
+
+            [Argument(ArgumentType.AtMostOnce,
+                HelpText = "Specifies whether the values are key values or numeric, only valid when loader is not specified and the type of data is not an idv.")]
+            public bool ValuesAsKeyType = true;
+        }
+
+        protected ValueMappingTransformer(IHostEnvironment env, IDataView lookupMap,
+            string keyColumn, string valueColumn, (string Input, string Output)[] columns)
+            : base(Contracts.CheckRef(env, nameof(env)).Register(nameof(ValueMappingTransformer)), columns)
+        {
+            env.CheckNonEmpty(keyColumn, nameof(keyColumn), "A key column must be specified when passing in an IDataView for the value mapping");
+            env.CheckNonEmpty(valueColumn, nameof(valueColumn), "A value column must be specified when passing in an IDataView for the value mapping");
+            _valueMap = CreateValueMapFromDataView(lookupMap, keyColumn, valueColumn);
+            int valueColumnIdx = 0;
+            env.Assert(lookupMap.Schema.TryGetColumnIndex(valueColumn, out valueColumnIdx));
+            _valueMetadata = CopyMetadata(lookupMap.Schema[valueColumnIdx].Metadata);
+
+            // Create the byte array of the original IDataView, this is used for saving out the data.
+            _dataView = GetBytesFromDataView(Host, lookupMap, keyColumn, valueColumn);
+        }
+
+        private Schema.Metadata CopyMetadata(Schema.Metadata metadata)
+        {
+            var meta = new MetadataBuilder();
+            meta.Add(metadata, x=> true);
```

Comment:
>x= [](start = 31, length = 2)

Ctrl+K+D

Reply:
That function is deleted, but I went ahead and formatted the whole file.

---
In reply to: [243494833](https://github.com/dotnet/machinelearning/pull/1710#discussion_r243494833) [](ancestors = 243494833)